dss_metrics/minio_rest_collector.py

```diff
@@ -52,90 +58,81 @@ def __init__(self, configs, whitelist_patterns, filter=False):
-        self.TYPE = 'minio'
+        self.TYPE = 'minio_rest'
+        self.logger = logging.getLogger(metrics.APP_NAME)
+
+    def collect(self):
+        self.logger.debug("--- MINIO REST COLLECTOR ---")
 
-    def poll_statistics(self, metrics_data_buffer, exit_flag):
-        cluster_endpoint_map = self.get_miniocluster_endpoint_map()
+        cluster_endpoint_map = utils.get_miniocluster_endpoint_map(self.mc,
+                                                                   self.logger)
+        cluster_endpoint_map_items = cluster_endpoint_map.items()
 
-        while True:
-            if exit_flag.is_set():
-                break
-            for _, endpts in cluster_endpoint_map.items():
-                if not endpts or len(endpts) == 0:
-                    continue
+        if (not cluster_endpoint_map_items or
+           len(cluster_endpoint_map_items) == 0):
+            self.logger.error("No MINIO endpoints found")
+            raise ValueError("No MINIO endpoints found")
 
-                minio_endpoint = endpts.pop()
-                miniocluster_id = self.get_minio_cluster_uuid(minio_endpoint)
+        all_minio_endpoints = []
+        for _, endpts in cluster_endpoint_map_items:
+            all_minio_endpoints.extend(list(endpts))
+
+        try:
+            for minio_endpoint in all_minio_endpoints:
+                miniocluster_id = utils.get_minio_cluster_uuid(
+                    minio_endpoint, self.url_prefix,
+                    self.cluster_id_url_suffix)
                 minio_metrics = self.get_minio_metrics_from_endpoint(
                     minio_endpoint)
 
+                if not miniocluster_id or not minio_metrics:
+                    self.logger.error("Failed to retrieve MINIO metadata")
+                    raise ValueError("Failed to retrieve MINIO metadata")
+
                 tags = {}
                 tags['cluster_id'] = self.cluster_id
                 tags['target_id'] = socket.gethostname()
                 tags['minio_id'] = miniocluster_id
+                tags['minio_endpoint'] = minio_endpoint
                 tags['type'] = self.TYPE
 
-                for metric in minio_metrics:
-                    if self.filter and not self.check_whitelist_key(metric[0]):
+                for key, value in minio_metrics:
+                    if (self.filter and not
+                            self.check_whitelist_key(key)):
                         continue
-                    metrics_data_buffer.append(
-                        metrics.MetricInfo(
-                            metric[0], metric[0], metric[1], tags, time.time())
+
+                    # check if scope applies to metric
+                    scope = self.get_metric_scope(key)
+                    if scope:
+                        tags['scope'] = scope
+
+                    name = key
+                    metric = Metric(name, key, 'gauge')
+                    metric.add_sample(
```

Comment:
this kind of code just increases the lines decreasing the readability.
If the logic is too complicated, makes sense to have each lines for arguments